crates/cli/src/job.rs

```diff
@@ -55,15 +55,25 @@ impl CacheDirectory {
     }
 }
 
+#[derive(Debug, Default, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[serde(tag = "type")]
+pub enum TriggerOn {
+    #[default]
+    None,
+    Branches(Vec<String>),
+    All,
+}
+
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 #[serde(tag = "type")]
 pub enum Trigger {
     Options {
         #[serde(default)]
-        push: Vec<String>,
+        push: TriggerOn,
```

Comment:
Can this be an `Option<TriggerOn>` as opposed to `#[serde(default)]` with the `None` type